stdlib/src/memory/unsafe_pointer.mojo

```diff
@@ -250,7 +247,7 @@ struct UnsafePointer[
         Returns:
             An offset pointer.
         """
-        return self + (-offset)
+        return self.offset(-offset)
```

Comment:
**Question** Does it work if we keep this line as `self + (-offset)`?

I understand this change is a bit of a simplification because it avoids an extra "hop" through `__add__` before going to `offset`. But for style and consistency reasons, representing pointer offsetting using `+` is preferred.

Reply:
Yea, it works. I simply didn't know that `__add__` is preferred. Will revert this change.

Reply:
No worries at all :)

(The fact that `UnsafePointer.offset` exists at the moment is a slight temporary state; it will likely get removed in favor of having only one way (`UnsafePointer.__add__`) to do offsetting.)

Reply:
Just a thought: why don't we simply remove `__add__` and `__sub__`, and only leave `offset` which is easier in method chaining (`(p + n)[]` looks bad)?